src/main/java/me/namuhuchutong/step3/Car.java

```diff
@@ -0,0 +1,37 @@
+package me.namuhuchutong.step3;
+
+public class Car {
+
+    private int position;
+    private final RacingRule racingRule;
+
+    public static Car from(Car car) {
+        return new Car(car.getRacingRule(), car.getPosition());
+    }
+
+    public Car(RacingRule racingRule, int position) {
+        validateCarPosition(position);
+        this.position = position;
+        this.racingRule = racingRule;
+    }
+
+    private void validateCarPosition(int position) {
+        if (position < 0) {
+            throw new IllegalArgumentException("자동차 위치는 음수일 수 없습니다. - " + position);
+        }
+    }
+
+    public void move() {
+        if (racingRule.isSatisfiedRule()) {
+            this.position += 1;
+        }
+    }
```

Comment:
```suggestion
    public void move(RacingRule rule) {
        if (rule.isSatisfiedRule()) {
            this.position += 1;
        }
    }
```
RacingRule이라는 인터페이스를 만드셨네요 ㅎㅎ
이렇게 수정된다면 move를 직접 테스트할 수 있을 것 같아요